fix(web): handle invalid locale parameters in i18n config

## Problem

Crawler requests to malformed URLs like `/favicon.ico/blog` cause errors in production. Next.js treats `favicon.ico` as a locale parameter, and `next-intl` attempts to load `./messages/favicon.ico.json` which doesn't exist.

**Sentry Issue:** WEB-3 - `Error: Cannot find module './favicon.ico.json'`

## Root Cause

In `apps/web/i18n.ts`, the dynamic import happens before locale validation:

```typescript
export default getRequestConfig(async ({ locale }) => {
  const resolvedLocale = locale || defaultLocale;
  return {
    locale: resolvedLocale,
    messages: (await import(`./messages/${resolvedLocale}.json`)).default,
  };
});
```

When `locale` is `favicon.ico`, it bypasses the `|| defaultLocale` check (since it's not falsy) and tries to import a non-existent file.

## Solution

Add locale validation **before** the dynamic import:

```typescript
const resolvedLocale =
  locale && locales.includes(locale as Locale) ? locale : defaultLocale;
```

This ensures invalid locales fall back to the default locale instead of causing import errors.

## Files to Change

- `apps/web/i18n.ts` - Add locale validation before dynamic import

--- a/turbo/apps/web/i18n.ts
+++ b/turbo/apps/web/i18n.ts
@@ -16,8 +16,10 @@ export const languageNames: Record<Locale, string> = {
 };
 
 export default getRequestConfig(async ({ locale }) => {
-  // Fallback to default locale if undefined
-  const resolvedLocale = locale || defaultLocale;
+  // Validate locale and fallback to default if invalid
+  // This prevents errors from malformed URLs like /favicon.ico/blog
+  const resolvedLocale =
+    locale && locales.includes(locale as Locale) ? locale : defaultLocale;
 
   return {
     locale: resolvedLocale,

